integration/query_bitwise_test.go

```diff
@@ -15,41 +15,517 @@
+			err: mongo.CommandError{
+				Code:    2,
+				Name:    "BadValue",
+				Message: "value takes an Array, a number, or a BinData but received: $bitsAnyClear: \"123\"",
+			},
+		},
+		"Int32": {
+			filter: int32(2),
+			expectedIDs: []any{
+				"binary-empty",
+				"double-negative-zero", "double-zero",
+				"int32-min", "int32-zero",
+				"int64-min", "int64-zero",
+			},
+		},
+		"Int32NegativeValue": {
+			filter: int32(-1),
+			err: mongo.CommandError{
+				Code:    9,
+				Name:    "FailedToParse",
+				Message: "Expected a positive number in: $bitsAnyClear: -1",
+			},
+		},
+		"Int64": {
+			filter: math.MaxInt64,
+			expectedIDs: []any{
+				"binary", "binary-empty",
+				"double-negative-zero", "double-whole", "double-zero",
+				"int32", "int32-max", "int32-min", "int32-zero",
+				"int64", "int64-min", "int64-zero",
+			},
+		},
+		"Int64NegativeValue": {
+			filter: int64(-1),
+			err: mongo.CommandError{
+				Code:    9,
+				Name:    "FailedToParse",
+				Message: "Expected a positive number in: $bitsAnyClear: -1",
+			},
+		},
+
+		"Array": {
+			filter: primitive.A{1, 5},
+			expectedIDs: []any{
+				"binary-empty",
+				"double-negative-zero", "double-zero",
+				"int32-min", "int32-zero",
+				"int64-min", "int64-zero",
+			},
+		},
+		"ArrayNegativeBitPositionValue": {
+			filter: primitive.A{-1},
+			err: mongo.CommandError{
+				Code:    2,
+				Name:    "BadValue",
+				Message: "bit positions must be >= 0 but got: 0: -1",
+			},
+		},
+		"ArrayBadValue": {
+			filter: primitive.A{"123"},
+			err: mongo.CommandError{
+				Code:    2,
+				Name:    "BadValue",
+				Message: `bit positions must be an integer but got: 0: "123"`,
+			},
+		},
+	} {
+		name, tc := name, tc
+		t.Run(name, func(t *testing.T) {
+			t.Parallel()
+
+			filter := bson.D{{"value", bson.D{{"$bitsAnyClear", tc.filter}}}}
+			cursor, err := collection.Find(ctx, filter, options.Find().SetSort(bson.D{{"_id", 1}}))
+			if tc.err.Code != 0 {
+				require.Nil(t, tc.expectedIDs)
+				AssertEqualError(t, tc.err, err)
+				return
+			}
+			require.NoError(t, err)
+
+			var actual []bson.D
+			err = cursor.All(ctx, &actual)
+			require.NoError(t, err)
+			assert.Equal(t, tc.expectedIDs, CollectIDs(t, actual))
+		})
+	}
+}
+
+func TestQueryBitwiseAnySet(t *testing.T) {
+	t.Parallel()
+	ctx, collection := setup(t, shareddata.Scalars)
+
+	for name, tc := range map[string]struct {
+		filter      any
+		expectedIDs []any
+		err         mongo.CommandError
+	}{
+		"Float": {
+			filter: 1.2,
+			err: mongo.CommandError{
+				Code:    9,
+				Name:    "FailedToParse",
+				Message: "Expected an integer: $bitsAnySet: 1.2",
+			},
+		},
+		"FloatWhole": {
+			filter: 2.0,
 			expectedIDs: []any{
-				"binary-empty", "double-negative-zero", "double-zero",
-				"int32-min", "int32-zero", "int64-min", "int64-zero",
+				"binary",
+				"double-whole",
+				"int32", "int32-max",
+				"int64", "int64-max",
+			},
+		},
+		"FloatNegativeValue": {
+			filter: int32(-1),
```

Comment:
That's not float. Also, there is no "float" at all; BSON and we call it "double"

Reply:
Fixed